apps/dashboard/src/components/workflow-editor/steps/edit-step-sidebar.tsx

```diff
@@ -36,24 +49,30 @@ export const EditStepSidebar = () => {
             transition={transitionSetting}
           />
         </SheetOverlay>
-        <motion.div
-          initial={{
-            x: '100%',
-          }}
-          animate={{
-            x: 0,
-          }}
-          exit={{
-            x: '100%',
-          }}
-          transition={transitionSetting}
-          className={
-            'bg-background fixed inset-y-0 right-0 z-50 flex h-full w-3/4 flex-col border-l shadow-lg sm:max-w-[600px]'
-          }
-        >
-          {/* TODO: show loading indicator */}
-          {workflow && step && stepType && <StepEditor workflow={workflow} step={step} stepType={stepType} />}
-        </motion.div>
+        <SheetContentBase asChild onInteractOutside={handleCloseSidebar} onEscapeKeyDown={handleCloseSidebar}>
```

Comment:
The `SheetContentBase` is the component that allows to handle the clicks on the drawer overlay and ESC button clicks. But the thing is that it requires the use of the `SheetTitle, SheetDescription` under the hood, but the current implementation doesn't use it. So I had to hide these two visually.
We might decide to refactor that in the future.

Reply:
Please add a TODO in the code.